Packages/com.unity.inputsystem/InputSystem/Plugins/Switch/NPad.cs

```diff
@@ -484,17 +484,46 @@ protected override void RefreshConfiguration()
                 m_NpadId = command.npadId;
                 m_Orientation = command.orientation;
                 m_StyleMask = command.styleMask;
-                ReadNNColorIntoJoyConColor(ref m_LeftControllerColor, command.colorLeftMain, command.colorLeftSub);
-                ReadNNColorIntoJoyConColor(ref m_RightControllerColor, command.colorRightMain, command.colorRightSub);
+                m_LeftControllerColor = NNColorToJoyConColor(command.colorLeftMain, command.colorLeftSub);
+                m_RightControllerColor = NNColorToJoyConColor(command.colorRightMain, command.colorRightSub);
+
+                switch (m_Orientation)
+                {
+                    case Orientation.Vertical:
+                        InputSystem.SetDeviceUsage(this, CommonUsages.Vertical);
+                        break;
+
+                    case Orientation.Horizontal:
+                        InputSystem.SetDeviceUsage(this, CommonUsages.Horizontal);
+                        break;
+                }
+
+                if (m_StyleMask == NpadStyle.JoyDual)
+                {
+                    InputSystem.AddDeviceUsage(this, CommonUsages.RightHand);
+                    InputSystem.AddDeviceUsage(this, CommonUsages.LeftHand);
+                }
+                else if (m_StyleMask == NpadStyle.JoyRight)
+                {
+                    InputSystem.AddDeviceUsage(this, CommonUsages.RightHand);
+                }
+                else if (m_StyleMask == NpadStyle.JoyLeft)
+                {
+                    InputSystem.AddDeviceUsage(this, CommonUsages.LeftHand);
+                }
```

Comment:
Looking at this again, the code here shouldn't use `SetDeviceUsage` but should instead rely entirely on `AddDeviceUsage` and `RemoveDeviceUsage` in order to preserve any usages the user may have set.

Reply:
Thanks, Rene. Martin noticed me that these changes should be in an NDA repo, so I'll fix it as you pointed out in there.

Reply:
Thank you @yebata. I'm sorry the changes didn't make it before the split-off. I kept having it on my list but then didn't get it done before Switch was moved out.